scripts/i18next-dynamic-keys-plugin.test.js

```diff
@@ -0,0 +1,314 @@
+        '  Expanded 3 dynamic keys from summary-log:failure.${...}'
+      )
+    })
+
+    it('should handle nested parent keys', () => {
+      // Mock en.json with nested structure
+      readFileSync.mockReturnValue(
+        JSON.stringify({
+          errors: {
+            validation: {
+              REQUIRED: 'Field is required',
+              INVALID: 'Field is invalid'
+            }
+          }
+        })
+      )
+
+      // Template literal: `common:errors.validation.${type}`
+      const expression = {
+        type: 'TemplateLiteral',
+        quasis: [
+          {
+            cooked: 'common:errors.validation.',
+            raw: 'common:errors.validation.'
+          },
+          { cooked: '', raw: '' }
+        ],
+        expressions: [{ type: 'Identifier', value: 'type' }]
+      }
+
+      const keys = plugin.extractKeysFromExpression(expression, {}, mockLogger)
+
+      expect(keys).toStrictEqual([
+        'common:errors.validation.REQUIRED',
+        'common:errors.validation.INVALID'
+      ])
+    })
+
+    it('should cache JSON files to avoid repeated reads', () => {
+      readFileSync.mockReturnValue(
+        JSON.stringify({
+          failure: {
+            CODE_A: 'Error A'
+          }
+        })
+      )
+
+      const expression = {
+        type: 'TemplateLiteral',
+        quasis: [
+          { cooked: 'summary-log:failure.', raw: 'summary-log:failure.' },
+          { cooked: '', raw: '' }
+        ],
+        expressions: [{ type: 'Identifier', value: 'code' }]
+      }
+
+      // Call twice with same namespace
+      plugin.extractKeysFromExpression(expression, {}, mockLogger)
+      plugin.extractKeysFromExpression(expression, {}, mockLogger)
+
+      // Should only read file once (cached on second call)
+      expect(readFileSync).toHaveBeenCalledTimes(1)
+    })
+
+    it('should throw error for missing en.json', () => {
+      readFileSync.mockImplementation(() => {
+        throw new Error('ENOENT: no such file or directory')
+      })
+
+      const expression = {
+        type: 'TemplateLiteral',
+        quasis: [
+          {
+            cooked: 'nonexistent:failure.',
+            raw: 'nonexistent:failure.'
+          },
+          { cooked: '', raw: '' }
+        ],
+        expressions: [{ type: 'Identifier', value: 'code' }]
+      }
+
+      expect(() => {
+        plugin.extractKeysFromExpression(expression, {}, mockLogger)
+      }).toThrowError(/ENOENT/)
+    })
+
+    it('should throw error for malformed JSON', () => {
+      readFileSync.mockReturnValue('{ invalid json }')
+
+      const expression = {
+        type: 'TemplateLiteral',
+        quasis: [
+          { cooked: 'namespace:failure.', raw: 'namespace:failure.' },
+          { cooked: '', raw: '' }
+        ],
+        expressions: [{ type: 'Identifier', value: 'code' }]
+      }
+
+      expect(() => {
+        plugin.extractKeysFromExpression(expression, {}, mockLogger)
+      }).toThrowError()
```

Comment:
You can add this to fix the linter warning:

```suggestion
      }).toThrowError(/Expected property name/)
```